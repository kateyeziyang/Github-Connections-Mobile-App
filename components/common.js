import * as React from 'react';
import { Text, View } from 'react-native';

export const detectInput = (extra, pageName) => {
  if (Object.keys(extra).length === 0) {
    console.log(pageName+'receive empty input.');
    return (
      <View><Text>Sorry, some error happened.</Text></View>
    )
  }
  if (typeof extra.name === 'undefined'
    && typeof extra.repositories === 'undefined'
    && typeof extra.followers === 'undefined'
    && typeof extra.following === 'undefined' 
    ) {
    console.log('Something wrong with '+pageName+' input.');
    console.log(extra);
    return (
      <View><Text>Sorry, something wrong with data.</Text></View>
    )
  }
  return true;
};
