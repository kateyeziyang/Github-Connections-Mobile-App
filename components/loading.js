import * as React from 'react';
import { Text, View } from 'react-native';

/**
 * Default screen when github haven't return anything.
 * @return A waiting page
*/
export function LoadingScreen() {
  return (
    <View>
      <Text>Loading...</Text>
    </View>
  );
}
