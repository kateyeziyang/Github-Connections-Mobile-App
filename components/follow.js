import * as React from 'react';
import { Text, View, Image, FlatList, StyleSheet } from 'react-native';
import { detectInput } from './common'

export function FollowScreen({ navigation, extra, isFollower }) {
  const detectResult = detectInput(extra, 'FollowScreen');
  if (detectResult !== true) {
    return detectResult;
  }
  let myData;
  if (isFollower) {
    myData=extra.followers.nodes;
  } else {
    myData=extra.following.nodes
  }
  return (
    <View style={styles.container}>
      <FlatList
        data={myData}
        renderItem={
          ({item}) =>
          <View>
            <Image source={{uri: item.avatarUrl}} style={styles.image}/>
            <Text style={styles.field}>
              Name: <Text style={styles.justText}>{item.name}</Text>{'\n'}
              Username:
              <Text
                onPress={()=>
                  navigation.push('Profile',
                    {uid: item.login, followPush: true}
                  )
                }
                style={styles.justText}>{item.login}
              </Text>
              {'\n'}{'\n'}
            </Text>
          </View>
        }
       keyExtractor={(item, index) => index.toString()}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  field: {
    color: 'steelblue',
    fontWeight: 'bold',
    fontSize: 20,
  },
  justText: {
    color: 'black',
    fontWeight: 'normal',
    fontSize: 18,
    lineHeight: 35,
  },
  image: {
    width: 150,
    height: 150,
    borderRadius: 150 / 2,
    overflow: "hidden",
    borderWidth: 3,
    borderColor: "white"
  },
  container: {
   flex: 1,
   alignItems: 'center',
   paddingTop: 22
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});
