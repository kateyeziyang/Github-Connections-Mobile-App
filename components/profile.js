import * as React from 'react';
import { Button, Text, StyleSheet, View, Image, ScrollView } from 'react-native';
import { detectInput } from './common'

/**
 * Screen when github information.
 * @param navigation Automatic parameter for navigating
 * @param extra Contains json reponse
 * @return A page contains user information
*/
export function ProfileScreen({ navigation, extra, doPush }) {
  const detectResult = detectInput(extra, 'ProfileScreen');
  if (detectResult !== true) {
    return detectResult;
  }

  let pic = {uri: extra.avatarUrl};
  let repos = extra.repositories;
  let repoCount = "Public Repos count: "+repos.totalCount;
  let followersCount = "Followers count: "+extra.followers.totalCount;
  let followingCount = "Following count: "+extra.following.totalCount;

  if (doPush === true) {
    return (
      <View style={{ flex: 1, alignItems: 'center', padding: 20}}>
        <Image source={pic} style={styles.image}/>
        <View style={{alignSelf: 'flex-start', padding: 20}}>
          <Text style={styles.field}>
          Name: <Text style={styles.justText}>{extra.name}</Text>{'\n'}
          Username: <Text style={styles.justText}>{extra.login}</Text>{'\n'}
          Bio: <Text style={styles.justText}>{extra.bio}</Text>{'\n'}
          Website: <Text style={styles.justText}>{extra.websiteUrl}</Text>{'\n'}
          Email: <Text style={styles.justText}>{extra.email}</Text>
          </Text>
          <Button
            title = {repoCount}
            onPress={() => navigation.push('Repo', {uid: extra.login})}
          />
          <Button
            title = {followersCount}
            onPress={() => navigation.push('Followers', {uid: extra.login, isFollower1: true})}
          />
          <Button
            title = {followingCount}
            onPress={() => navigation.push('Following', {uid: extra.login, isFollower1: false})}
          />
          <Text style={styles.field}>
          Profile creation date: <Text style={styles.justText}>{extra.updatedAt}</Text>
          </Text>
        </View>
      </View>
    );
  } else {
    return (
      <View style={{ flex: 1, alignItems: 'center', padding: 20}}>
        <Image source={pic} style={styles.image}/>
        <ScrollView style={{ flex: 1, padding: 20}}>
          <Text style={styles.field}>
          Name: <Text style={styles.justText}>{extra.name}</Text>{'\n'}
          Username: <Text style={styles.justText}>{extra.login}</Text>{'\n'}
          Bio: <Text style={styles.justText}>{extra.bio}</Text>{'\n'}
          Website: <Text style={styles.justText}>{extra.websiteUrl}</Text>{'\n'}
          Email: <Text style={styles.justText}>{extra.email}</Text>
          </Text>
          <Button
            title = {repoCount}
            onPress={() => navigation.navigate('Repo')}
          />
          <Button
            title = {followersCount}
            onPress={() => navigation.navigate('Followers', {addtionalData: extra, useAdditonal: true})}
          />
          <Button
            title = {followingCount}
            onPress={() => navigation.navigate('Following', {addtionalData: extra, useAdditonal: true})}
          />
          <Text style={styles.field}>
          Profile creation date: <Text style={styles.justText}>{extra.updatedAt}</Text>
          </Text>
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  field: {
    color: 'steelblue',
    fontWeight: 'bold',
    fontSize: 20,
  },
  justText: {
    color: 'black',
    fontWeight: 'normal',
    fontSize: 18,
    lineHeight: 35,
  },
  image: {
    width: 150,
    height: 150,
    borderRadius: 150 / 2,
    overflow: "hidden",
    borderWidth: 3,
    borderColor: "white"
  },
});
