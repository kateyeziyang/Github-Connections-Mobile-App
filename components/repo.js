import * as React from 'react';
import { FlatList, StyleSheet, Text, View } from 'react-native';
import { detectInput } from './common'

/**
 * Screen with github repository info.
 * @param navigation Automatic parameter for navigating
 * @param extra Contains json reponse
 * @return Screen with github repository info
*/
export function RepoScreen({ extra, navigation }) {
  const detectResult = detectInput(extra, 'RepoScreen');
  if (detectResult !== true) {
    return detectResult;
  }
  return (
    <View style={styles.container}>
      <Text style={styles.item}><Text style={styles.field}>Name</Text> | Owner | Description </Text>
      <FlatList
        data={extra.repositories.nodes}
        renderItem={
          ({item}) =>
          <Text style={styles.item}>
          <Text style={styles.field}>{item.name} </Text>
          |
          <Text
            onPress={()=>
              navigation.push('Profile',
                {uid: item.owner.login, followPush: true}
              )
            }
            style={styles.justText}>
           {item.owner.login} |
          </Text>
           {item.description}
          </Text>
        }
       keyExtractor={(item, index) => index.toString()}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  field: {
    color: 'steelblue',
    fontWeight: 'bold',
    fontSize: 20,
  },
  container: {
   flex: 1,
   paddingTop: 22
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
})
