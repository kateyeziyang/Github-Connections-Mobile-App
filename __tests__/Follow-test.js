/**
 * @format
 */

import 'react-native';
import React from 'react';
import { FollowScreen } from '../components/follow';
import CONSTANTS from '../constants';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { api } from '../api';

beforeEach(() => {
  fetch.resetMocks();
});

it ('invalid response', () => {
  fetch.mockResponseOnce(JSON.stringify({id:1}));
  const nav = { navigate: jest.fn() };
  var ext;
  try {
    ext = api('posts');
  } catch (e) {
    console.log(e);
  }

  const tree = renderer
    .create(<FollowScreen navigation = {nav} extra = {ext} />).toJSON();

  expect(tree).toMatchSnapshot();
})

it ('empty response', () => {
  const nav = { navigate: jest.fn() };
  var ext;
  ext = {};
  const tree = renderer
    .create(<FollowScreen navigation = {nav} extra = {ext} />).toJSON();

  expect(tree).toMatchSnapshot();
});


it('renders correctly', () => {
  const nav = { navigate: jest.fn() };
  const ext = CONSTANTS.ARR;
  const tree = renderer
    .create(<FollowScreen navigation = {nav} extra = {ext} />).toJSON();

  expect(tree).toMatchSnapshot();
});
