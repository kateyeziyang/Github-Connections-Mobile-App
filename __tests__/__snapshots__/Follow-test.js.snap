// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty response 1`] = `
<View>
  <Text>
    Sorry, some error happened.
  </Text>
</View>
`;

exports[`invalid response 1`] = `
<View>
  <Text>
    Sorry, something wrong with data.
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 22,
    }
  }
>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
